build/azure-pipelines/product-build.yml

```diff
@@ -134,14 +135,30 @@ stages:
     steps:
     - template: darwin/product-build-darwin.yml
 
+- stage: macOSARM64
+  dependsOn:
+  - Compile
+  condition: and(succeeded(), eq(variables['VSCODE_COMPILE_ONLY'], 'false'))
+  pool:
+    vmImage: macOS-latest
```

Comment:
For now we are cross-compiling but when ci supports native compilation, it will be using a different pool. Hence this is written as a separate stage.